### PR TITLE
Fix supervisor panic when tkns and publicapi Service is nil

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Review only commit https://github.com/gitpod-io/gitpod/pull/15274/commits/e081051ad939278d68faa63642fd1a8c38c362fe

Test also

```golang
package main

import (
	"fmt"
)

var t TestInterface
var t2 TestInterface

func main() {
	t = NewImplement()
	t.Echo("hello")

	t2 = NewImplement2()
	t2.Echo("hello")
}

type TestInterface interface {
	Echo(something string)
}

type Implement struct {
	hello *string
}

func NewImplement() *Implement {
	return nil
}

func NewImplement2() TestInterface {
	return nil
}

var _ TestInterface = (*Implement)(nil)

// Echo implements TestInterface
func (s *Implement) Echo(something string) {
	if s == nil {
		fmt.Println("s is nil")
		return
	}
	fmt.Println(*s.hello, something)
}

```

## Related Issue(s)

Fixes #

## How to test

- Open workspace with repo that will buildimage i.e. https://github.com/gitpod-io/openvscode-server
- Image build should success
- Workpsaces open should works

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by mustard-mh** [comment]
Tested

|ImageBuild|WorkspaceStart|
|-|-|
|SCR-20221212-pni|SCR-20221212-pr9|

File: components/common-go/experiments/types.go [hunk 1]
```diff
@@ -32,10 +37,23 @@ type Attributes struct {
 // If the environment contains CONFIGCAT_SDK_KEY value, it vill be used to construct a ConfigCat client.
 // Otherwise, it returns a client which always returns the default value. This client is used for Self-Hosted installations.
 func NewClient() Client {
+	gitpodHost := os.Getenv("GITPOD_HOST")
```
Comment (akosyakov) [inline components/common-go/experiments/types.go:40]: @mustard-mh it is a internal secret of supervisor which is a subject to change, why don't pass it as an argument?

**Review by akosyakov — APPROVED** [review]
LGTM

File: components/common-go/experiments/types.go [hunk 2]
```diff
@@ -32,10 +37,23 @@ type Attributes struct {
...
 		return NewAlwaysReturningDefaultValueClient()
 	}
-
-	return newConfigCatClient(sdkKey)
+	return newConfigCatClient(configcat.Config{
+		SDKKey:       sdkKey,
+		PollInterval: 3 * time.Minute,
+		HTTPTimeout:  3 * time.Second,
+		Logger:       &configCatLogger{log.Log},
+	})
```
Comment (easyCZ) [inline components/common-go/experiments/types.go:58]: there's duplication in the config that could be reduced.

**Review by easyCZ — APPROVED** [review]
approving to unblock, see comments for style/improvements

**Pull request merged**